src/main/java/uk/gov/hmcts/ccd/domain/model/common/CatalogueResponse.java

```diff
@@ -0,0 +1,74 @@
+package uk.gov.hmcts.ccd.domain.model.common;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder;
+
+@JsonDeserialize(builder = CatalogueResponse.Builder.class)
+public class CatalogueResponse<T> {
+
+    private final String code;
+    private final String message;
+    private final T details;
+
+    public CatalogueResponse() {
+        this(null,null,null);
+    }
+
+    private CatalogueResponse(final String code, final String message, final T details) {
+        this.code = code;
+        this.message = message;
+        this.details = details;
+    }
+
+    private CatalogueResponse(final String code, final String message) {
+        this(code,message,null);
+    }
+
+    public CatalogueResponse(final CatalogueResponseElement catalogueResponseElement, final T details) {
+        this(catalogueResponseElement.getCode(), catalogueResponseElement.getMessage(), details);
+    }
+
+    public CatalogueResponse(final CatalogueResponseElement catalogueResponseElement) {
+        this(catalogueResponseElement, null);
+    }
```

Comment:
the whole point of having a builder is to avoid having telescoping constructor pattern, which is what you have here anyway. We should get rid of all these overloaded constructors.

Reply:
http://www.developerdave.co.uk/2016/05/using-the-builder-pattern-in-jackson/